### PR TITLE
Dynamic raster VS vector decision

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
This PR depends on 
- [ ] https://github.com/CartoDB/cartodb.js/pull/1624.
- [ ] https://github.com/CartoDB/Windshaft-cartodb/issues/664

### CODE REVIEW

File: src/api/create-vis.js [hunk 1]
```diff
@@ -75,6 +76,13 @@ var loadVizJSON = function (el, visModel, vizjsonData, options) {
     layerSelectorEnabled = options.layerSelectorEnabled;
   }
 
+  var renderMode = RenderModes.AUTO;
```
Comment (alonsogarciapablo) [inline src/api/create-vis.js]: cc: @donflopez @javisantana I kept `vector` as the option for CartoDB.js for now. But I introduced the concept of render modes internally.

**Comment by alonsogarciapablo** [comment]
Retest this please!

File: src/api/create-vis.js (hunk 1 above)
Comment (javisantana) [inline src/api/create-vis.js]: :+1:

File: src/geo/leaflet/leaflet-layer-view-factory.js [hunk 2]
```diff
@@ -1,24 +1,54 @@
...
+    return true;
   }
-  return new LeafletCartoDBLayerGroupView(layerGroupModel, mapModel);
+
+  if (mapRenderMode === RenderModes.RASTER) {
+    return false;
+  }
+
+  // mapRenderMode === RenderModes.AUTO
+  return mapModel.getNumberOfFeatures() < MAX_NUMBER_OF_FEATURES_FOR_WEBGL &&
```
Comment (alonsogarciapablo) [inline src/geo/leaflet/leaflet-layer-view-factory.js]: :point_up:  @donflopez @javisantana here's the code that decides whether to use vector or raster when renderMode is auto.

**Comment by alonsogarciapablo** [comment]
retest this please

File: src/geo/leaflet/leaflet-layer-view-factory.js [hunk 3]
```diff
@@ -1,24 +1,54 @@
...
 var LeafletCartoDBLayerGroupView = require('./leaflet-cartodb-layer-group-view');
 var LeafletTorqueLayerView = require('./leaflet-torque-layer-view');
 var LeafletCartoDBWebglLayerGroupView = require('./leaflet-cartodb-webgl-layer-group-view');
+var TangramCartoCSS = require('tangram-cartocss');
+var RenderModes = require('../../vis/render-modes');
 
-var LayerGroupViewConstructor = function (layerGroupModel, mapModel, options) {
-  if (options.vector) {
-    return new LeafletCartoDBWebglLayerGroupView(layerGroupModel, mapModel);
+var MAX_NUMBER_OF_FEATURES_FOR_WEBGL = 1000;
```
Comment (donflopez) [inline src/geo/leaflet/leaflet-layer-view-factory.js]: It was 10e4 not 10e2, isn't it?

**Pull request merged**